linode/firewall/framework_models.go

```diff
@@ -0,0 +1,168 @@
+package firewall
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/linode/linodego"
+)
+
+// FirewallModel describes the Terraform resource data model to match the
+// resource schema.
+type FirewallModel struct {
+	ID             types.Int64  `tfsdk:"id"`
+	Label          types.String `tfsdk:"label"`
+	Tags           types.Set    `tfsdk:"tags"`
+	Disabled       types.Bool   `tfsdk:"disabled"`
+	Inbound        types.List   `tfsdk:"inbound"`
+	InboundPolicy  types.String `tfsdk:"inbound_policy"`
+	Outbound       types.List   `tfsdk:"outbound"`
+	OutboundPolicy types.String `tfsdk:"outbound_policy"`
+	Linodes        types.Set    `tfsdk:"linodes"`
+	Devices        types.List   `tfsdk:"devices"`
+	Status         types.String `tfsdk:"status"`
+}
+
+func (data *FirewallModel) parseComputedAttributes(
+	ctx context.Context,
+	firewall *linodego.Firewall,
+	rules *linodego.FirewallRuleSet,
+	devices []linodego.FirewallDevice,
+) diag.Diagnostics {
+	data.ID = types.Int64Value(int64(firewall.ID))
+	data.Disabled = types.BoolValue(firewall.Status == linodego.FirewallDisabled)
+	data.Status = types.StringValue(string(firewall.Status))
+
+	linodes, diag := types.SetValueFrom(ctx, types.Int64Type, parseFirewallLinodes(devices))
```

Comment:
Can you consider rename all `diag` to be `diags` if they are `type Diagnostics []Diagnostic`? I think naming them as `diag` can make it confused with a single `Diagnostic` object, right?

Reply:
Fixed. Thanks for catching it.